src/cli/self_update/windows.rs

```diff
@@ -575,18 +564,17 @@ fn _remove_from_path(old_path: Vec<u16>, path_str: Vec<u16>) -> Option<Vec<u16>>
     if new_path.last() == Some(&(b';' as u16)) {
         new_path.pop();
     }
-    Some(new_path)
+    Some(HSTRING::from_wide(&new_path))
 }
 
-fn _with_path_cargo_home_bin<F>(f: F, process: &Process) -> Result<Option<Vec<u16>>>
+fn _with_path_cargo_home_bin<F>(f: F, process: &Process) -> Result<Option<HSTRING>>
 where
-    F: FnOnce(Vec<u16>, Vec<u16>) -> Option<Vec<u16>>,
+    F: FnOnce(HSTRING, HSTRING) -> Option<HSTRING>,
 {
     let windows_path = get_windows_path_var()?;
     let mut path_str = process.cargo_home()?;
     path_str.push("bin");
-    Ok(windows_path
-        .and_then(|old_path| f(old_path, OsString::from(path_str).encode_wide().collect())))
+    Ok(windows_path.and_then(|old_path| f(old_path, HSTRING::from(path_str.into_os_string()))))
```

Comment:
There seems to be a `From<&Path> for HSTRING`, so I think we can make this `HSTRING::from(&path_str)`?

Reply:
Implicit deref coercion cannot be used there because `HSTRING::from()` method includes a lot of overloads.